WcaOnRails/app/controllers/competitions_controller.rb

```diff
@@ -96,6 +100,14 @@ def index
       @competitions = @competitions.select { |competition| competition.has_events_with_ids?(params[:event_ids]) }
     end
 
+    unless params[:status] == "all"
+      if params[:status] == "warning"
+        @competitions = @competitions.select { |competition| competition.pending_results_or_report(14) }
+      else
+        @competitions = @competitions.select { |competition| competition.pending_results_or_report(21) }
```

Comment:
> How are we going to remember to keep this code in sync with report_class and results_class?

I created model variables for this, but...honestly, looks UGLY :P haha

> Also, what do you plan to do if someone changes results_class so it does something different than report_class?

You mean if the params are no longer the same for them? I think that would require a regulation change (see [1c1](https://www.worldcubeassociation.org/regulations/#1c1) and [1c3](https://www.worldcubeassociation.org/regulations/#1c3)) so I'm not too worried.